invenio/modules/relationships/engines/sqlalchemy.py

```diff
@@ -0,0 +1,145 @@
+        :param outwards: boolean
+            Count the edges that are coming out from this node.
+        :param link_type: string
+            Count only this type of edges. Optional.
+
+        :returns: int
+            Number of edges.
+        """
+        return len(self.edges(inwards, outwards, link_type))
+
+    def edges(self, inwards=True, outwards=True, link_type=None,
+              loops_doubled=True):
+        """Get all the edges of this node.
+
+        There are no guarantees on the order of the result.
+
+        :param inwards: boolean
+            Get the edges that are coming in to this node.
+        :param outwards: boolean
+            Get the edges that are coming out from this node.
+        :param link_type: string
+            Type of the edges. Optional.
+        :param loops_doubled: boolean
+            If true, return every loop twice.
+
+        :returns: list
+            Edges belonging to this node.
+        """
+        edges = SQLAlchemyRelationship.query
+        if inwards and outwards:
+            edges = edges.filter(db.or_(SQLAlchemyRelationship.id_to ==
+                                        self.node_id,
+                                        SQLAlchemyRelationship.id_from ==
+                                        self.node_id))
+        elif inwards:
+            edges = edges.filter_by(id_to=self.node_id)
+        elif outwards:
+            edges = edges.filter_by(id_from=self.node_id)
+        else:
+            return []
+        if link_type:
+            edges = edges.filter_by(link_type=link_type)
+
+        if inwards and outwards and loops_doubled:
+            # Special case for loops. Second query needed.
+            reflexives = SQLAlchemyRelationship.query.filter(
+                db.and_(SQLAlchemyRelationship.id_to == self.node_id,
+                        SQLAlchemyRelationship.id_from == self.node_id))
+            if link_type:
+                reflexives = reflexives.filter(link_type=link_type)
+            return edges.all() + reflexives.all()
+
+        return edges.all()
+
+    def _loop(self, edges):
+        """Check if the list of edges contain loop."""
+        for edge in edges:
+            if (edge.id_from == edge.id_to and edge.storagename_from ==
+                    edge.storagename_to):
+                return True
+        return False
+
+    def neighbours(self, inwards=True, outwards=True, link_type=None):
+        """Get all the nodes which are neighbours of this node.
+
+        :param inwards: boolean
+            Get the vertices that are sources for this node.
+        :param outwards: boolean
+            Get the vertices that are destinations from this node.
+        :param link_type: string
+            Type of the edges. Optional.
+
+        :returns: set
+            Nodes that are neighbours of this node.
+        """
+        edges = self.edges(inwards, outwards, link_type, loops_doubled=False)
+        set1 = {SQLAlchemyNode(id=node.id_from,
```

Comment:
is it Python 2.6 friendly? we should check that nobody is planning to use *Invenio 2.0* on machines with Python < 2.7